test/model.indexes.test.js

```diff
@@ -666,5 +666,24 @@ describe('model', function() {
         ]);
       });
     });
+    it('should prevent collation on text indexes (gh-10044)', function() {
+      return co(function*() {
+        yield db.db.collection('User').drop().catch(() => {});
+
+        const userSchema = new mongoose.Schema({ username: String }, {
+          collation: {
+            locale: 'en',
+            strength: 2
+          }
+        });
+        userSchema.index({ username: 'text' }, { unique: true });
+        const User = db.model('User', userSchema, 'User');
+
+        yield User.init();
+        const indexes = yield User.listIndexes();
+        assert.ok(!indexes[1].collation);
+        yield User.collection.drop();
```

Comment:
Doesn't look like this `drop()` call is necessary. Does it cause tests to fail if we remove it?

Reply:
does not, will remove